### ISSUE TITLE
Counter type from statsdreceiver being dropped by prometheusexporter

### ISSUE THREAD

**Issue by locmai** [open-telemetry/opentelemetry-collector-contrib#4153]
**Describe the bug**
The counting-type metric from StatsD receiver will always be dropped by the prometheusexporter

**Steps to reproduce**
- Set the config..yaml as the one below the Config section
- Build from source and run the binary / Run on k8s cluster
- Send a simple `c` metric: `echo "locmai.test:30|c" | nc -w 1 -v -u 0.0.0.0 8127`

**What did you expect to see?**
The locmai_test metric being exported in the Prometheus Exporter http://0.0.0.0:9090

**What did you see instead?**
No metric be exported from the prometheusexporter, check the metrics.json and the metric is exported there.

**What version did you use?**
Version: v0.29.0

**What config did you use?**
Config:

```yaml
receivers:
  statsd:
    endpoint: "0.0.0.0:8127"
    aggregation_interval: 5s
    enable_metric_type: true
    timer_histogram_mapping:
      - statsd_type: "histogram"
        observer_type: "summary"
      - statsd_type: "timing"
        observer_type: "summary"
processors:
  batch:
exporters:
  prometheus:
    endpoint: "0.0.0.0:9090"
    metric_expiration: 180m
  file:
    path: ./metrics.json
service:
  pipelines:
    metrics:
      receivers: [statsd]
      processors: [batch]
      exporters: [prometheus,file]
```

**Environment**
OS: CentOS, Kubernetes, MacOS, 

**Additional context**
This was due to the logic for the statsdreceiver that set SetAggregationTemporality(pdata.AggregationTemporalityDelta)
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/139e7dc1549ea364d7a8e33fcad2593e33d0f623/receiver/statsdreceiver/protocol/metric_translator.go#L32

And the prometheusexporter is dropping any metric without the AggregationTemporalityCumulative

https://github.com/open-telemetry/opentelemetry-collector/blob/cc41009d95166f0c5ab9a9bddbab9ec903ed163d/exporter/prometheusexporter/accumulator.go#L198

It only check for AggregationTemporalityCumulative, not AggregationTemporalityDelta.

So for the proposal solution:
Fix these lines from prometheusexporter to check the other AggregationTemporality types as well before dropping the metric.

**Comment by locmai** [comment]
I'm re-opening this since my last fix was off the spec of statsd receiver, re-trying with the hope of fixing this one at the prometheusexporter end.

**Comment by locmai** [comment]
Another attempt from @nabam I found in the PR list #9006 - this one also updated the `accumulateDoubleHistogram` function but doesn't include the fix for statsd receiver side in #9919
